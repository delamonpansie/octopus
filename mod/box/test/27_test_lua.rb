#!/usr/bin/ruby
# encoding: ASCII

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
object_space[0].enabled = 1
object_space[0].index[0].type = "HASH"
object_space[0].index[0].unique = 1
object_space[0].index[0].key_field[0].fieldno = 0
object_space[0].index[0].key_field[0].type = "STR"

object_space[0].index[1].type = "TREE"
object_space[0].index[1].unique = 0
object_space[0].index[1].key_field[0].fieldno = 0
object_space[0].index[1].key_field[0].type = "STR"
object_space[0].index[1].key_field[1].fieldno = 1
object_space[0].index[1].key_field[1].type = "NUM"
object_space[0].index[1].key_field[2].fieldno = 2
object_space[0].index[1].key_field[2].type = "STR"


object_space[1].enabled = 1
object_space[1].index[0].type = "HASH"
object_space[1].index[0].unique = 1
object_space[1].index[0].key_field[0].fieldno = 0
object_space[1].index[0].key_field[0].type = "NUM32"
object_space[1].index[1].type = "HASH"
object_space[1].index[1].unique = 1
object_space[1].index[1].key_field[0].fieldno = 1
object_space[1].index[1].key_field[0].type = "NUM32"
object_space[1].index[2].type = "HASH"
object_space[1].index[2].unique = 1
object_space[1].index[2].key_field[0].fieldno = 2
object_space[1].index[2].key_field[0].type = "NUM64"
object_space[1].index[3].type = "TREE"
object_space[1].index[3].unique = 0
object_space[1].index[3].key_field[0].fieldno = 0
object_space[1].index[3].key_field[0].type = "NUM32"


object_space[2].enabled = 1
object_space[2].index[0].type = "TREE"
object_space[2].index[0].unique = 1
object_space[2].index[0].key_field[0].fieldno = 0
object_space[2].index[0].key_field[0].type = "NUM32"
object_space[2].index[0].key_field[1].fieldno = 1
object_space[2].index[0].key_field[1].type = "NUM32"


EOD
  end
end

Env.connect_eval do
  100.times {|i| insert [i.to_s, i, i.to_s] }
  insert ["\0\0\0\0", "\0\0\0\0", "\0\0\0\0\0\0\0\0"], :object_space => 1
  insert ["\0\0\0\0", "\0\0\0\0", "----"], :object_space => 2

  (1..9).each do |i|
    lua "user_proc.test#{i}"
  end
end
