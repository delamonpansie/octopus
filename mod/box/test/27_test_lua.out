# box.insert(["0", 0, "0"])
1

# box.insert(["1", 1, "1"])
1

# box.insert(["2", 2, "2"])
1

# box.insert(["3", 3, "3"])
1

# box.insert(["4", 4, "4"])
1

# box.insert(["5", 5, "5"])
1

# box.insert(["6", 6, "6"])
1

# box.insert(["7", 7, "7"])
1

# box.insert(["8", 8, "8"])
1

# box.insert(["9", 9, "9"])
1

# box.insert(["10", 10, "10"])
1

# box.insert(["11", 11, "11"])
1

# box.insert(["12", 12, "12"])
1

# box.insert(["13", 13, "13"])
1

# box.insert(["14", 14, "14"])
1

# box.insert(["15", 15, "15"])
1

# box.insert(["16", 16, "16"])
1

# box.insert(["17", 17, "17"])
1

# box.insert(["18", 18, "18"])
1

# box.insert(["19", 19, "19"])
1

# box.insert(["20", 20, "20"])
1

# box.insert(["21", 21, "21"])
1

# box.insert(["22", 22, "22"])
1

# box.insert(["23", 23, "23"])
1

# box.insert(["24", 24, "24"])
1

# box.insert(["25", 25, "25"])
1

# box.insert(["26", 26, "26"])
1

# box.insert(["27", 27, "27"])
1

# box.insert(["28", 28, "28"])
1

# box.insert(["29", 29, "29"])
1

# box.insert(["30", 30, "30"])
1

# box.insert(["31", 31, "31"])
1

# box.insert(["32", 32, "32"])
1

# box.insert(["33", 33, "33"])
1

# box.insert(["34", 34, "34"])
1

# box.insert(["35", 35, "35"])
1

# box.insert(["36", 36, "36"])
1

# box.insert(["37", 37, "37"])
1

# box.insert(["38", 38, "38"])
1

# box.insert(["39", 39, "39"])
1

# box.insert(["40", 40, "40"])
1

# box.insert(["41", 41, "41"])
1

# box.insert(["42", 42, "42"])
1

# box.insert(["43", 43, "43"])
1

# box.insert(["44", 44, "44"])
1

# box.insert(["45", 45, "45"])
1

# box.insert(["46", 46, "46"])
1

# box.insert(["47", 47, "47"])
1

# box.insert(["48", 48, "48"])
1

# box.insert(["49", 49, "49"])
1

# box.insert(["50", 50, "50"])
1

# box.insert(["51", 51, "51"])
1

# box.insert(["52", 52, "52"])
1

# box.insert(["53", 53, "53"])
1

# box.insert(["54", 54, "54"])
1

# box.insert(["55", 55, "55"])
1

# box.insert(["56", 56, "56"])
1

# box.insert(["57", 57, "57"])
1

# box.insert(["58", 58, "58"])
1

# box.insert(["59", 59, "59"])
1

# box.insert(["60", 60, "60"])
1

# box.insert(["61", 61, "61"])
1

# box.insert(["62", 62, "62"])
1

# box.insert(["63", 63, "63"])
1

# box.insert(["64", 64, "64"])
1

# box.insert(["65", 65, "65"])
1

# box.insert(["66", 66, "66"])
1

# box.insert(["67", 67, "67"])
1

# box.insert(["68", 68, "68"])
1

# box.insert(["69", 69, "69"])
1

# box.insert(["70", 70, "70"])
1

# box.insert(["71", 71, "71"])
1

# box.insert(["72", 72, "72"])
1

# box.insert(["73", 73, "73"])
1

# box.insert(["74", 74, "74"])
1

# box.insert(["75", 75, "75"])
1

# box.insert(["76", 76, "76"])
1

# box.insert(["77", 77, "77"])
1

# box.insert(["78", 78, "78"])
1

# box.insert(["79", 79, "79"])
1

# box.insert(["80", 80, "80"])
1

# box.insert(["81", 81, "81"])
1

# box.insert(["82", 82, "82"])
1

# box.insert(["83", 83, "83"])
1

# box.insert(["84", 84, "84"])
1

# box.insert(["85", 85, "85"])
1

# box.insert(["86", 86, "86"])
1

# box.insert(["87", 87, "87"])
1

# box.insert(["88", 88, "88"])
1

# box.insert(["89", 89, "89"])
1

# box.insert(["90", 90, "90"])
1

# box.insert(["91", 91, "91"])
1

# box.insert(["92", 92, "92"])
1

# box.insert(["93", 93, "93"])
1

# box.insert(["94", 94, "94"])
1

# box.insert(["95", 95, "95"])
1

# box.insert(["96", 96, "96"])
1

# box.insert(["97", 97, "97"])
1

# box.insert(["98", 98, "98"])
1

# box.insert(["99", 99, "99"])
1

# box.insert(["\x00\x00\x00\x00", "\x00\x00\x00\x00", "\x00\x00\x00\x00\x00\x00\x00\x00"], {:object_space=>1})
1

# box.insert(["\x00\x00\x00\x00", "\x00\x00\x00\x00", "----"], {:object_space=>2})
1

# box.lua("user_proc.test1")
[["abc", "defg", "foobar"], ["abc", "defg"], ["abc"]]

# box.lua("user_proc.test2")
[["cdata<const struct BasicIndex *>: 0xPTR", "cdata<const struct BasicIndex *>: 0xPTR"]]

# box.lua("user_proc.test3")
[["cdata<const struct box_small_tuple *>: 0xPTR", "11", "11"]]

# box.lua("user_proc.test4")
[["300"]]

# box.lua("user_proc.test5")
[["\x00\x00\x00\x00", "\x00\x00\x00\x00", "\x00\x00\x00\x00\x00\x00\x00\x00"], ["\x00\x00\x00\x00", "\x00\x00\x00\x00", "\x00\x00\x00\x00\x00\x00\x00\x00"], ["\x00\x00\x00\x00", "\x00\x00\x00\x00", "\x00\x00\x00\x00\x00\x00\x00\x00"], ["\x00\x00\x00\x00", "\x00\x00\x00\x00", "\x00\x00\x00\x00\x00\x00\x00\x00"]]

# box.lua("user_proc.test6")
[["\x00\x00\x00\x00", "\x00\x00\x00\x00", "\x00\x00\x00\x00"]]

# box.lua("user_proc.test7")
[["99", "c\x00\x00\x00", "99"], ["99", "c\x00\x00\x00", "9999"], ["99", "c\x00\x00\x00", "9999"], ["nil"]]

# box.lua("user_proc.test8")
[["\x00\x00\x00\x00", "\x00\x00\x00\x00", "----"], ["\x00\x00\x00\x00", "\x00\x00\x00\x00", "9999"], ["\x00\x00\x00\x00", "\x00\x00\x00\x00", "9999"], ["nil"]]

# box.lua("user_proc.test9")
[["\x00\x00\x00\x00", "\x00\x00\x00\x00", "", "", ""], ["\x00\x00\x00\x00", "\x00\x00\x00\x00", "", "", ""], ["\x00\x00\x00\x00", "\x00\x00\x00\x00", "\x00\x00", "\x00\x00\x00\x00", "\x00\x00\x00\x00\x00\x00\x00\x00"], ["\x00\x00\x00\x00", "\x00\x00\x00\x00", "\x00\x00", "\x0E\x00\x00\x00", "c\x00\x00\x00\x00\x00\x00\x00"], ["\x00\x00\x00\x00", "\x00\x00\x00\x00", "\x00\x00", "\x0E\x00\x00\x00", "c\x00\x00\x00\x00\x00\x00\x00"]]

